src/sentry/notifications/notifications/base.py

```diff
@@ -180,14 +164,16 @@ def get_sentry_query_params(
         return f"?referrer={self.get_referrer(provider, recipient)}"
 
     def get_settings_url(self, recipient: Team | User, provider: ExternalProviders) -> str:
-        # settings url is dependant on the provider so we know which provider is sending them into Sentry
+        # Settings url is dependant on the provider so we know which provider is sending them into Sentry.
         if isinstance(recipient, Team):
             team = Team.objects.get(id=recipient.id)
-            url_str = f"/settings/{self.org_slug}/teams/{team.slug}/notifications/"
+            url_str = f"/settings/{self.organization.slug}/teams/{team.slug}/notifications/"
         else:
             url_str = "/settings/account/notifications/"
-            if self.fine_tuning_key:
-                url_str += f"{self.fine_tuning_key}/"
+            if self.notification_setting_type:
+                fine_tuning_key = get_notification_setting_type_name(self.notification_setting_type)
```

Comment:
nit: If we refactor the way `get_notification_setting_type_name()` functions to allow it to accept `str | None`, then return `None` if passed `None`, we could get rid of this nested if, or at least move it into the function.